lib/db2.js

```diff
@@ -50,6 +51,8 @@ function DB2(settings) {
     ';PORT=' + this.portnumber +
     ';PROTOCOL=' + this.protocol;
 
+  this.connStr = this.dsn || this.connStrGenerate;
+
   this.schema = this.username;
   if (settings.schema) {
     this.connStr += ';CurrentSchema=' + settings.schema;
```

Comment:
As far I am concerned, it's ok to leave the validation of DSN to the database driver and/or server. We can always implement the validation later, based on feedback from users. We are not validating URL values in other connectors either.